unittests/optiontests.py

```diff
@@ -246,6 +246,69 @@ def test_augment_set_sub(self):
         self.assertEqual(optstore.get_value_for(name), top_value)
         self.assertEqual(optstore.get_value_for(name, sub_name), set_value)
 
+    def test_build_to_host(self):
+        name = 'cpp_std'
+        def_value = 'c++98'
+        opt_value = 'c++17'
+        optstore = OptionStore(False)
+        co = UserComboOption(name,
+                             'C++ language standard to use',
+                             def_value,
+                             choices=['c++98', 'c++11', 'c++14', 'c++17', 'c++20', 'c++23'],
+                             )
+        optstore.add_compiler_option('cpp', OptionKey(name), co)
+
+        cmd_line = {OptionKey(name): opt_value}
+        optstore.initialize_from_top_level_project_call({}, cmd_line, {})
+        self.assertEqual(optstore.get_value_object_and_value_for(OptionKey(name, machine=MachineChoice.BUILD))[1], opt_value)
+        self.assertEqual(optstore.get_value(OptionKey(name, machine=MachineChoice.BUILD)), opt_value)
+        self.assertEqual(optstore.get_value_for(OptionKey(name, machine=MachineChoice.BUILD)), opt_value)
+
+    def test_build_to_host_subproject(self):
+        name = 'cpp_std'
+        def_value = 'c++98'
+        opt_value = 'c++17'
+        subp = 'subp'
+        optstore = OptionStore(False)
+        co = UserComboOption(name,
+                             'C++ language standard to use',
+                             def_value,
+                             choices=['c++98', 'c++11', 'c++14', 'c++17', 'c++20', 'c++23'],
+                             )
+        optstore.add_compiler_option('cpp', OptionKey(name), co)
+
+        spcall = {OptionKey(name): opt_value}
+        optstore.initialize_from_top_level_project_call({}, {}, {})
+        optstore.initialize_from_subproject_call(subp, spcall, {}, {}, {})
+        self.assertEqual(optstore.get_value_object_and_value_for(OptionKey(name, subproject=subp,
+                                                                           machine=MachineChoice.BUILD))[1], opt_value)
+        self.assertEqual(optstore.get_value(OptionKey(name, subproject=subp, machine=MachineChoice.BUILD)), opt_value)
+        self.assertEqual(optstore.get_value_for(OptionKey(name, subproject=subp, machine=MachineChoice.BUILD)), opt_value)
+
+    def test_build_to_host_cross(self):
+        name = 'cpp_std'
+        def_value = 'c++98'
+        opt_value = 'c++17'
+        optstore = OptionStore(True)
+        for key in [OptionKey(name), OptionKey(name, machine=MachineChoice.BUILD)]:
```

Comment:
It might be clearer to store this as:
```python
key = OptionKey('cpp_std')
```
The  you can access the build variant as `key.as_build()`

Just a thought